# 4. 반복문

반복문은 **특정 조건에 따라 똑같은 기능을 반복해서 수행**하는 구문입니다.   
의료 현장에서는 여러 환자의 데이터를 처리하거나, 검사 결과를 순차적으로 분석할 때 반복문이 필수적입니다.

<aside>

💡 **반복문 리딩 스킬**

반복문을 **"각각에 대해"**로 읽으면 코드 이해가 쉬워집니다!

</aside>



## 🔁 for문 - 정해진 횟수만큼

for문은 **정해진 범위나 리스트의 요소들을 순회하며 하나씩 처리**할 때 사용합니다.

In [ ]:
# 환자 명단 순회
patient_names = ["김환자", "이환자", "박환자", "최환자"]  # 오늘 진료 환자 명단

for patient_name in patient_names:
    print(f"{patient_name}님, 진료실로 들어오세요!")

김환자님, 진료실로 들어오세요!
이환자님, 진료실로 들어오세요!
박환자님, 진료실로 들어오세요!
최환자님, 진료실로 들어오세요!


반복문이 없다면

In [ ]:
# ---- loop 1 ----#
print(patient_names[0])

# ---- loop 1 ----#
print(patient_names[1])

# ---- loop 1 ----#
print(patient_names[2])

# ---- loop 1 ----#
print(patient_names[3])


김환자
이환자
박환자
최환자


## ❓문제 1. 약물 복용 안내

처방된 약물들에 대해 복용 안내를 출력하세요.  
아래와 같은 형태로 출력
```
"""
[약물명]을 식후에 복용하세요
...
"""
```



In [ ]:
medicines = ["타이레놀", "소화제", "비타민", "항생제"]

In [ ]:
# 여기에 코드 작성


In [ ]:
# @title 정답
medicines = ["타이레놀", "소화제", "비타민", "항생제"]

for medicine in medicines:
    print(f"{medicine}을 식후에 복용하세요")

타이레놀을 식후에 복용하세요
소화제을 식후에 복용하세요
비타민을 식후에 복용하세요
항생제을 식후에 복용하세요


## 🔍 유용한 반복문 함수들

| 함수 | 용도 | 반환값 | 예시 |
| --- | --- | --- | --- |
| `range()` | 연속된 숫자 생성 | 숫자 시퀀스 | `range(5)` → 0,1,2,3,4 |
| `enumerate()` | 인덱스와 값 함께 | (인덱스, 값) 튜플 | `enumerate(['a','b'])` → (0,'a'), (1,'b') |
| `zip()` | 여러 리스트 동시 순회 | 각 리스트의 요소들을 묶은 튜플 | `zip([1,2], ['a','b'])` → (1,'a'), (2,'b') |



### enumerate() - 인덱스와 값을 함께

`enumerate()`는 **인덱스(순서)와 값을 함께** 가져올 때 사용합니다.



In [ ]:
# 기본 사용법 - 환자 대기 순번
patients = ["김환자", "이환자", "박환자", "최환자"]  # 대기 환자 명단

for index, patient_name in enumerate(patients):
    print(f"{index + 1}번째 대기: {patient_name}")


1번째 대기: 김환자
2번째 대기: 이환자
3번째 대기: 박환자
4번째 대기: 최환자


In [ ]:
# 시작 번호 지정하기 - 병실 번호 배정
patients = ["김환자", "이환자", "박환자"]  # 입원 환자 명단

print("=== 병실 배정 현황 ===")
for room_number, patient_name in enumerate(patients, 101):  # 101호부터 시작
    print(f"{room_number}호: {patient_name}")


=== 병실 배정 현황 ===
101호: 김환자
102호: 이환자
103호: 박환자


In [ ]:
# 활용하기 - 처방전 목록
prescriptions = ["해열제", "항생제", "진통제", "비타민"]  # 처방 약물 목록

print("오늘의 처방전")
for num, medicine in enumerate(prescriptions, start=1):
    print(f"{num}. {medicine}")

오늘의 처방전
1. 해열제
2. 항생제
3. 진통제
4. 비타민


### ❓문제 2. 검사 항목 번호 매기기

검사 항목에 번호를 매겨서 출력하세요.  
아래와 같은 형태로 출력
```
"""
1. [검사항목]
2. [검사항목]
...
"""
```



In [ ]:
test_items = ["혈압측정", "혈당검사", "체중측정", "시력검사"]

In [ ]:
# 여기에 코드 작성


In [ ]:
# @title 정답
test_items = ["혈압측정", "혈당검사", "체중측정", "시력검사"]

for num, item in enumerate(test_items, 1):
    print(f"{num}. {item}")

1. 혈압측정
2. 혈당검사
3. 체중측정
4. 시력검사


### zip() - 여러 리스트를 동시에

`zip()`은 **여러 개의 리스트를 동시에 순회**할 때 사용합니다.




In [ ]:
# 기본 사용법 - 환자 기본 정보
patient_names = ["김환자", "이환자", "박환자"]  # 환자 이름 목록
patient_ages = [65, 45, 32]  # 환자 나이 목록
departments = ["내과", "외과", "소아과"]  # 담당 진료과 목록

for data in zip(patient_names, patient_ages, departments):
    print(data) # 튜플로 묶임

('김환자', 65, '내과')
('이환자', 45, '외과')
('박환자', 32, '소아과')


In [ ]:
# 튜플로 묶여있는 데이터를 언패킹으로 각각 꺼내오기
for name, age, department in zip(patient_names, patient_ages, departments):
    print(f"{name}님({age}세)은 {department}에서 진료받습니다.")


김환자님(65세)은 내과에서 진료받습니다.
이환자님(45세)은 외과에서 진료받습니다.
박환자님(32세)은 소아과에서 진료받습니다.


In [ ]:
# 활용하기 - 검사 항목과 결과
test_items = ["혈압", "혈당", "콜레스테롤", "체중"]  # 검사 항목명
test_results = [120, 95, 180, 70]  # 검사 결과값
units = ["mmHg", "mg/dL", "mg/dL", "kg"]  # 측정 단위

print("🔬 검사 결과")
for item, result, unit in zip(test_items, test_results, units):
    print(f"{item}: {result} {unit}")


🔬 검사 결과
혈압: 120 mmHg
혈당: 95 mg/dL
콜레스테롤: 180 mg/dL
체중: 70 kg


In [ ]:
# 길이가 다른 리스트 - 짧은 것에 맞춰짐
symptoms = ["발열", "기침", "두통"]  # 환자 증상 목록
treatments = ["해열제", "기침약", "진통제", "항생제"]  # 가능한 치료법

for symptom, treatment in zip(symptoms, treatments):
    print(f"{symptom} → {treatment}")

# 출력: 3개만 출력됨 (symptoms가 3개라서)

발열 → 해열제
기침 → 기침약
두통 → 진통제


### ❓문제 3. 환자 처방 매칭

zip()을 활용하여 환자 명단과 처방 약물, 복용법을 함께 출력하세요.
아래와 같은 형태로 출력
```
"""
환자명: 처방약물 (복용법)
...
"""
```


In [ ]:
patient_names = ["김환자", "이환자", "박환자", "최환자"]  # 환자 명단
prescribed_drugs = ["해열제", "항생제", "진통제", "비타민"]  # 처방 약물
dosages = ["1일 3회", "1일 2회", "1일 1회", "1일 1회"]  # 복용법


In [ ]:
# 여기에 코드 작성


In [ ]:
# @title 정답
patient_names = ["김환자", "이환자", "박환자", "최환자"]
prescribed_drugs = ["해열제", "항생제", "진통제", "비타민"]
dosages = ["1일 3회", "1일 2회", "1일 1회", "1일 1회"]

for name, drug, dosage in zip(patient_names, prescribed_drugs, dosages):
    print(f"{name}: {drug} ({dosage})")


김환자: 해열제 (1일 3회)
이환자: 항생제 (1일 2회)
박환자: 진통제 (1일 1회)
최환자: 비타민 (1일 1회)


### range() 함수 활용

range() 함수는 **연속된 숫자를 생성**하는 함수입니다.

**range() 함수의 인자**

- `range(끝)`: 0부터 끝-1까지
- `range(시작, 끝)`: 시작부터 끝-1까지
- `range(시작, 끝, 간격)`: 시작부터 끝-1까지 간격만큼 증가



#### 리스트로 변환하기




In [ ]:
# range(끝): 0부터 끝-1까지 병실 번호 생성
room_numbers = list(range(5))  # 0~4번 격리실
print(f"격리실 번호: {room_numbers}")


격리실 번호: [0, 1, 2, 3, 4]


In [ ]:
# range(시작, 끝): 시작부터 끝-1까지 일반 병실 번호
general_rooms = list(range(101, 111))  # 101~110호 일반실
print(f"일반실 번호: {general_rooms}")


일반실 번호: [101, 102, 103, 104, 105, 106, 107, 108, 109, 110]


In [ ]:
# range(시작, 끝, 간격): 간격만큼 증가하는 VIP실 번호
vip_rooms = list(range(201, 211, 2))  # 201, 203, 205, 207, 209호
print(f"VIP실 번호: {vip_rooms}")


VIP실 번호: [201, 203, 205, 207, 209]


In [ ]:
# 거꾸로 카운트 - 수술 대기 순서
surgery_queue = list(range(10, 0, -1))  # 10번부터 1번까지
print(f"수술 대기 순서: {surgery_queue}")

수술 대기 순서: [10, 9, 8, 7, 6, 5, 4, 3, 2, 1]


#### 반복문에서 사용하기

In [ ]:
# range(끝): 0부터 끝-1까지 - 기본 검사 항목
for i in range(5):  # 0, 1, 2, 3, 4
    print(f"{i + 1}번째 기본 검사를 진행합니다.")


1번째 기본 검사를 진행합니다.
2번째 기본 검사를 진행합니다.
3번째 기본 검사를 진행합니다.
4번째 기본 검사를 진행합니다.
5번째 기본 검사를 진행합니다.


In [ ]:
# range(시작, 끝): 시작부터 끝-1까지 - 병실 순회
for room_num in range(101, 106):
    print(f"{room_num}호 병실을 점검합니다.")


101호 병실을 점검합니다.
102호 병실을 점검합니다.
103호 병실을 점검합니다.
104호 병실을 점검합니다.
105호 병실을 점검합니다.


In [ ]:
# range(시작, 끝, 간격): 간격만큼 증가 - 2시간마다 투약
for hour in range(8, 21, 2):
    print(f"{hour}시에 약물을 투여합니다.")


8시에 약물을 투여합니다.
10시에 약물을 투여합니다.
12시에 약물을 투여합니다.
14시에 약물을 투여합니다.
16시에 약물을 투여합니다.
18시에 약물을 투여합니다.
20시에 약물을 투여합니다.


In [ ]:
# 거꾸로 카운트 - 마취 카운트다운
print("=== 마취 시작 ===")
for i in range(10, 0, -1):  # 10, 9, 8, ..., 1
    print(f"카운트다운: {i}")
print("마취 완료!")


=== 마취 시작 ===
카운트다운: 10
카운트다운: 9
카운트다운: 8
카운트다운: 7
카운트다운: 6
카운트다운: 5
카운트다운: 4
카운트다운: 3
카운트다운: 2
카운트다운: 1
마취 완료!


In [ ]:
# 활용하기 - 혈압 측정 기록
patient_name = "김환자"  # 환자 이름
base_systolic = 120  # 기준 수축기 혈압

print(f"=== {patient_name} 혈압 측정 기록 ===")
for day in range(1, 8):  # 1주일간 측정
    systolic = base_systolic + (day * 2)  # 매일 2씩 증가 가정
    print(f"{day}일차: {systolic} mmHg")

=== 김환자 혈압 측정 기록 ===
1일차: 122 mmHg
2일차: 124 mmHg
3일차: 126 mmHg
4일차: 128 mmHg
5일차: 130 mmHg
6일차: 132 mmHg
7일차: 134 mmHg


### ❓문제 4. 병실 점검

range를 활용하여 101호부터 105호까지의 병실을 점검하는 메시지를 출력하세요.  
아래와 같은 형태로 출력
```
"""
[병실번호]호 병실을 점검합니다
...
"""
```



In [ ]:
# 여기에 코드 작성


In [ ]:
# @title 정답
for room in range(101, 106):
    print(f"{room}호 병실을 점검합니다")

101호 병실을 점검합니다
102호 병실을 점검합니다
103호 병실을 점검합니다
104호 병실을 점검합니다
105호 병실을 점검합니다


## 📊 딕셔너리 반복



In [ ]:
# 환자 기본 정보
patient_info = {
    "name": "김환자",  # 환자 이름
    "age": 65,  # 환자 나이
    "blood_type": "A+",  # 혈액형
    "department": "내과"  # 담당 진료과
}


In [ ]:
# 키만 반복 - 환자 정보 항목 확인
for key in patient_info:  # 기본적으로 키만 반복
    print(f"정보 항목: {key}")


정보 항목: name
정보 항목: age
정보 항목: blood_type
정보 항목: department


In [ ]:
# 키만 명시적으로 반복
for key in patient_info.keys():
    print(f"데이터 키: {key}")


데이터 키: name
데이터 키: age
데이터 키: blood_type
데이터 키: department


In [ ]:
# 값만 반복 - 환자 정보 내용만
for value in patient_info.values():
    print(f"정보 내용: {value}")
# 키와 값 동시 반복 - 완전한 환자 정보
for key, value in patient_info.items():
    print(f"{key}: {value}")

정보 내용: 김환자
정보 내용: 65
정보 내용: A+
정보 내용: 내과
name: 김환자
age: 65
blood_type: A+
department: 내과


## ❓문제 5. 환자 정보 출력

환자의 기본 정보를 ```[항목]: [값]``` 형태로 모두 출력하세요.




In [ ]:
patient = {
    "이름": "홍환자",
    "나이": 42,
    "혈액형": "B+",
    "진료과": "정형외과"
}

In [ ]:
# 여기에 코드 작성


In [ ]:
# @title 정답
patient = {
    "이름": "홍환자",
    "나이": 42,
    "혈액형": "B+",
    "진료과": "정형외과"
}

for key, value in patient.items():
    print(f"{key}: {value}")

이름: 홍환자
나이: 42
혈액형: B+
진료과: 정형외과


## ♾️ while문 - 조건이 맞을 때까지

while문은 **조건이 참인 동안 계속 반복**합니다.


In [ ]:
# 기본 while문
count = 1

while count <= 5:
    print(f"현재 숫자: {count}")
    count += 1  # count = count + 1 과 같음

print("카운트 완료!")

현재 숫자: 1
현재 숫자: 2
현재 숫자: 3
현재 숫자: 4
현재 숫자: 5
카운트 완료!


활용하기

In [ ]:
# 혈압 안정화 대기
systolic_pressure = 160  # 현재 수축기 혈압
target_pressure = 140  # 목표 혈압
hour_count = 0  # 경과 시간

print("=== 혈압 안정화 치료 ===")
while systolic_pressure > target_pressure:
    hour_count += 1
    systolic_pressure -= 5  # 시간당 5씩 감소 가정
    print(f"{hour_count}시간 후: {systolic_pressure} mmHg")

print(f"혈압 안정화 완료! (총 {hour_count}시간 소요)")

=== 혈압 안정화 치료 ===
1시간 후: 155 mmHg
2시간 후: 150 mmHg
3시간 후: 145 mmHg
4시간 후: 140 mmHg
혈압 안정화 완료! (총 4시간 소요)


## 🛑 break와 continue

- **break**: 반복문을 완전히 종료
- **continue**: 현재 반복을 건너뛰고 다음 반복으로


In [ ]:
# break - 특정 조건에서 반복 종료
for i in range(1, 11):
    if i == 5:
        print("5를 만났으니 종료!")
        break  # 반복문 완전 종료
    print(f"숫자: {i}")


숫자: 1
숫자: 2
숫자: 3
숫자: 4
5를 만났으니 종료!


In [ ]:
# continue - 특정 조건에서 건너뛰기
for i in range(1, 6):
    if i == 3:
        print("3은 건너뛰기!")
        continue  # 3일 때만 아래 코드 실행 안함
    print(f"숫자: {i}")


숫자: 1
숫자: 2
3은 건너뛰기!
숫자: 4
숫자: 5


활용하기

In [ ]:
# 통합 환자 데이터
patient_data = {
    "김환자": {"systolic": 120, "allergic": False},  # 수축기 혈압, 알레르기 여부
    "이환자": {"systolic": 145, "allergic": True},
    "박환자": {"systolic": 160, "allergic": False},
    "최환자": {"systolic": 135, "allergic": True},
    "정환자": {"systolic": 180, "allergic": False}
}

In [ ]:
# break - 위험 혈압 환자 첫 발견시 즉시 응급 처치
danger_threshold = 170  # 위험 혈압 기준값

print("=== 응급 혈압 환자 검색 ===")
for patient_name, data in patient_data.items():
    systolic = data["systolic"]  # 수축기 혈압
    print(f"{patient_name} 혈압 확인: {systolic} mmHg")

    if systolic >= danger_threshold:
        print(f"🚨 위험! {patient_name}의 혈압이 {systolic} mmHg입니다!")
        print("즉시 응급 처치를 시작합니다.")
        break  # 위험 환자 발견시 즉시 중단

    print(f"{patient_name} 정상 범위")

print("\n응급 혈압 환자 검색을 종료합니다.")


=== 응급 혈압 환자 검색 ===
김환자 혈압 확인: 120 mmHg
김환자 정상 범위
이환자 혈압 확인: 145 mmHg
이환자 정상 범위
박환자 혈압 확인: 160 mmHg
박환자 정상 범위
최환자 혈압 확인: 135 mmHg
최환자 정상 범위
정환자 혈압 확인: 180 mmHg
🚨 위험! 정환자의 혈압이 180 mmHg입니다!
즉시 응급 처치를 시작합니다.

응급 혈압 환자 검색을 종료합니다.


In [ ]:
print("=== 페니실린 처방 가능 환자 ===")
# continue 예제 - 알레르기 환자 제외하고 처방
for patient_name, data in patient_data.items():
    if data["allergic"]:  # 알레르기 여부 확인
        print(f"{patient_name}: 알레르기로 처방 제외")
        continue  # 알레르기 환자는 건너뛰기

    systolic = data["systolic"]
    print(f"{patient_name}: 페니실린 처방 가능 (혈압: {systolic} mmHg)")

=== 페니실린 처방 가능 환자 ===
김환자: 페니실린 처방 가능 (혈압: 120 mmHg)
이환자: 알레르기로 처방 제외
박환자: 페니실린 처방 가능 (혈압: 160 mmHg)
최환자: 알레르기로 처방 제외
정환자: 페니실린 처방 가능 (혈압: 180 mmHg)


## ❓문제 6. 응급환자 발견

환자들의 체온을 순서대로 확인하다가 38.5도 이상인 고열 환자를 발견하면
"응급환자 발견! 즉시 처치 필요" 출력하고 반복을 중단하세요.



In [ ]:
temperatures = [36.8, 37.2, 39.1, 36.5, 37.8]

In [ ]:
# 여기에 코드 작성


In [ ]:
# @title 정답
temperatures = [36.8, 37.2, 39.1, 36.5, 37.8]

for temp in temperatures:
    print(f"체온 확인: {temp}도")
    if temp >= 38.5:
        print("응급환자 발견! 즉시 처치 필요")
        break

체온 확인: 36.8도
체온 확인: 37.2도
체온 확인: 39.1도
응급환자 발견! 즉시 처치 필요


## ✨ 내포용법 (List Comprehension)

내포용법은 **리스트를 간단하게 만드는 방법**입니다.


In [ ]:
# 기본 방법 - 환자 이름을 정중한 호칭으로 변환
patient_names = ["김환자", "이환자", "박환자"]  # 환자 이름 목록
formal_names = []  # 정중한 호칭 목록

for name in patient_names:
    formal_names.append(name + "님")
print(f"기본 방법: {formal_names}")


기본 방법: ['김환자님', '이환자님', '박환자님']


In [ ]:
# 내포용법 사용
patient_names = ["김환자", "이환자", "박환자"]
formal_names = [name + "님" for name in patient_names]
print(f"내포용법: {formal_names}")


내포용법: ['김환자님', '이환자님', '박환자님']


In [ ]:
# 숫자 변환 - 체온을 화씨로 변환
celsius_temps = [36.5, 37.2, 38.5, 39.1]  # 섭씨 체온
fahrenheit_temps = [temp * 9/5 + 32 for temp in celsius_temps]
print(f"화씨 체온: {fahrenheit_temps}")


화씨 체온: [97.7, 98.96000000000001, 101.3, 102.38000000000001]


In [ ]:
# 문자열 처리 - 약물명을 대문자로
medicines = ["aspirin", "ibuprofen", "acetaminophen"]  # 약물명 목록
upper_medicines = [medicine.upper() for medicine in medicines]
print(f"대문자 약물명: {upper_medicines}")


대문자 약물명: ['ASPIRIN', 'IBUPROFEN', 'ACETAMINOPHEN']


In [ ]:
# 조건 포함 - 고혈압 환자만 선택
systolic_pressures = [110, 145, 130, 160, 125, 180]  # 수축기 혈압 목록
hypertension_cases = [pressure for pressure in systolic_pressures if pressure >= 140]
print(f"고혈압 수치: {hypertension_cases}")


고혈압 수치: [145, 160, 180]


In [ ]:
# 조건 포함 - 성인/미성년자 분류
patient_ages = [15, 25, 35, 10, 45, 8, 65]  # 환자 나이 목록
age_categories = ["성인" if age >= 18 else "미성년자" for age in patient_ages]
print(f"환자 분류: {age_categories}")

환자 분류: ['미성년자', '성인', '성인', '미성년자', '성인', '미성년자', '성인']


In [ ]:
# 복합 처리 - 정상 체온만 화씨로 변환
temperatures = [36.2, 38.5, 37.1, 39.2, 36.8]  # 체온 목록 (섭씨)
normal_temps_f = [temp * 9/5 + 32 for temp in temperatures if temp < 37.5]
print(f"정상 체온(화씨): {normal_temps_f}")


정상 체온(화씨): [97.16, 98.78, 98.24]


In [ ]:
# 길이 계산 - 약물명 글자 수
drugs = ["아스피린", "이부프로펜", "타이레놀", "펜타닐"]  # 약물명 목록
name_lengths = [len(drug) for drug in drugs]
print(f"약물명 길이: {name_lengths}")

약물명 길이: [4, 5, 4, 3]


## ❓문제 7. 정상 혈압 환자

혈압 리스트에서 정상 혈압(120 이하)인 환자의 혈압만 골라내는
새로운 리스트를 만드세요. (리스트 컴프리헨션 사용)



In [ ]:
blood_pressures = [110, 145, 118, 160, 125, 135]

In [ ]:
# 여기에 코드 작성


In [ ]:
# @title 정답
blood_pressures = [110, 145, 118, 160, 125, 135]

normal_pressures = [bp for bp in blood_pressures if bp <= 120]
print(f"정상 혈압: {normal_pressures}")

정상 혈압: [110, 118]


## 💊 실전 예제: 환자 검사 결과 분석 시스템



In [ ]:
# 환자 검사 결과 데이터
patient_results = [
    {"name": "김환자", "age": 65, "systolic": 145, "blood_sugar": 110},  # 환자별 검사 결과
    {"name": "이환자", "age": 45, "systolic": 130, "blood_sugar": 95},
    {"name": "박환자", "age": 32, "systolic": 120, "blood_sugar": 85},
    {"name": "최환자", "age": 70, "systolic": 160, "blood_sugar": 200}
]

In [ ]:
# 1. 모든 환자 기본 정보 출력
for patient in patient_results:
    name = patient["name"]  # 환자 이름
    age = patient["age"]  # 환자 나이
    systolic = patient["systolic"]  # 수축기 혈압
    blood_sugar = patient["blood_sugar"]  # 혈당 수치
    print(f"{name}({age}세): 혈압 {systolic}mmHg, 혈당 {blood_sugar}mg/dL")

김환자(65세): 혈압 145mmHg, 혈당 110mg/dL
이환자(45세): 혈압 130mmHg, 혈당 95mg/dL
박환자(32세): 혈압 120mmHg, 혈당 85mg/dL
최환자(70세): 혈압 160mmHg, 혈당 200mg/dL


In [ ]:
# 2. 고위험군 환자 찾기
high_risk_patients = []  # 고위험군 환자 이름 목록

for patient in patient_results:
    name = patient["name"]
    age = patient["age"]
    systolic = patient["systolic"]
    blood_sugar = patient["blood_sugar"]

    # 위험 요소 확인
    is_hypertension = systolic >= 140  # 고혈압 여부
    is_diabetes = blood_sugar >= 140  # 당뇨 여부
    is_elderly = age >= 65  # 고령자 여부

    # 2개 이상 조건 만족시 고위험군
    risk_factors = [is_hypertension, is_diabetes, is_elderly]
    if sum(risk_factors) >= 2: # sum: True의 갯수 합산
        high_risk_patients.append(name)

# 고위험군 환자 순번 출력
if high_risk_patients:
    for rank, patient_name in enumerate(high_risk_patients, 1):
        print(f"{rank}순위 고위험군: {patient_name}")
else:
    print("고위험군 환자가 없습니다.")

1순위 고위험군: 김환자
2순위 고위험군: 최환자


In [ ]:
# 3. 검사 항목별 평균 계산 (리스트 컴프리헨션 활용)
systolic_values = [patient["systolic"] for patient in patient_results]  # 모든 혈압 수치
blood_sugar_values = [patient["blood_sugar"] for patient in patient_results]  # 모든 혈당 수치

avg_systolic = sum(systolic_values) / len(systolic_values)  # 평균 혈압
avg_blood_sugar = sum(blood_sugar_values) / len(blood_sugar_values)  # 평균 혈당

print(f"평균 수축기 혈압: {avg_systolic:.1f} mmHg")
print(f"평균 혈당: {avg_blood_sugar:.1f} mg/dL")

평균 수축기 혈압: 138.8 mmHg
평균 혈당: 122.5 mg/dL


In [ ]:
# 4. 정상 범위 환자 비율
normal_bp_patients = [patient["name"] for patient in patient_results if patient["systolic"] < 140]
normal_bs_patients = [patient["name"] for patient in patient_results if patient["blood_sugar"] < 140]

print(f"\n정상 혈압 환자: {len(normal_bp_patients)}명 ({normal_bp_patients})")
print(f"정상 혈당 환자: {len(normal_bs_patients)}명 ({normal_bs_patients})")


정상 혈압 환자: 2명 (['이환자', '박환자'])
정상 혈당 환자: 3명 (['김환자', '이환자', '박환자'])
